api/Controllers/AnalysisMappingController.cs

```diff
@@ -0,0 +1,273 @@
+                inspectionDescription,
+                tagId
+            );
+            if (analysisMapping == null)
+            {
+                analysisMapping = await analysisMappingService.CreateAnalysisMapping(
+                    tagId,
+                    inspectionDescription,
+                    at
+                );
+            }
+            else
+            {
+                analysisMapping = await analysisMappingService.AddAnalysisTypeToMapping(
+                    analysisMapping.Id,
+                    at.Value
+                );
+            }
+
+            return Ok(analysisMapping);
+        }
+        catch (InvalidOperationException e)
+        {
+            logger.LogError(
+                e,
+                "Analysis mapping already exists for this tag and inspection description"
+            );
+            return BadRequest(
+                "Analysis mapping already exists for this tag and inspection description"
+            );
+        }
+        catch (ArgumentException)
+        {
+            logger.LogError("Analysis type already exists in analysis mapping");
+            return BadRequest("Analysis type already exists in analysis mapping");
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, "Error during creation of analysis mapping");
+            return StatusCode(
+                StatusCodes.Status500InternalServerError,
+                "An error occurred while creating the analysis mapping"
+            );
+        }
+    }
+
+    /// <summary>
+    /// Add a new analysis type to an existing analysis mapping
+    /// </summary>
+    [HttpPatch]
+    [Route("analysisMappingId/{analysisMappingId}/analysisType/{analysisType?}")]
+    [Authorize(Roles = Role.Any)]
+    [ProducesResponseType(typeof(AnalysisMapping), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<AnalysisMapping>> AddAnalysisTypeToMapping(
+        [FromRoute] string analysisMappingId,
+        [FromRoute] string? analysisType = null
+    )
+    {
+        try
+        {
+            var at = Analysis.TypeFromString(analysisType);
+            if (at == null)
+            {
+                return BadRequest("Invalid analysis type");
+            }
+
+            var analysisMapping = await analysisMappingService.AddAnalysisTypeToMapping(
+                analysisMappingId,
+                at.Value
+            );
+
+            return Ok(analysisMapping);
+        }
+        catch (ArgumentException)
+        {
+            logger.LogError("Could not add analysis type to mapping");
+            return BadRequest("Could not add analysis type to mapping");
+        }
```

Comment:
Maybe the error handling should be similar to the one above?

```suggestion
        catch (ArgumentException)
        {
            logger.LogError("Analysis type already exists in analysis mapping");
            return BadRequest("Analysis type already exists in analysis mapping");
        }
```

Reply:
Though it would be benefitial to have different messages for different functions for easier debugging. Also at the moment, the analysisMappingService can throw ArgumentException in several different scanarios, so the message is not neccessarily correct. Open to suggestions on message

Reply:
Updated